### DISCUSSION TITLE
Exclude an entity from generating migrations

### DISCUSSION THREAD

**Discussion by Zmaon** [mikro-orm/mikro-orm#5251] [General]
Hello,

In a project I need to use a few tables that are generated by a external process. I would like to be able to create entities to retrieve data from those tables, but to avoid migrations to be generated for them. Is that possible?

Thanks,

Simon

**Comment by B4nan** [comment]
Currently you can only ignore whole schemas in postgres, not specific tables, but that should be rather easy to support, will try to look into that.

**Reply by Zmaon** [reply]
That would be nice. I'm using the MySQL driver btw.

**Reply by B4nan** [reply]
Hmm I thought your use case is that you want to have some entities for some tables, and ignore the rest of the tables, but rereading this now, you said you want to ignore entity from migrations? Why is that? Does it produce some unwanted alter table statements? You don't have all the columns defined?

**Reply by Zmaon** [reply]
My use case is that I have this table (let's call it 'EXT') generated by an external process in the same MySQL database as the tables created by the mikro-orm migrations. To use EXT efficiently I created an entity for it.

Now if i run `mikro-orm:create` is generates a new migration for this table, from the entity. Thinking about it I realize that I can simply delete the migration while keeping the change on the snapshot and it should be ok, but each time I will modify the entity to reflect changes on this table it will create new migrations. I guess in my case it would be simpler if I had a way to tell in the entity that no migrations should be generated for this table. 

I understand that this may be an edge case though, and may not justify a change if no one else required such a thing.

**Reply by B4nan** [reply]
What I don't understand is why is your entity out of sync with the actual table? You don't want to model all its properties?

**Reply by Zmaon** [reply]
Well I may be doing something wrong, but here's what happens in my project:

1. So the table already exists in database, because an external process (an ETL) is creating and maintaining it. 
2. I create en entity mimicking the existing table, recreating all the properties. 
3. I run `mikro-orm migration:create` -> It creates a migration with a `CREATE TABLE` for this table. I guess this is because the table doesn't exist in the snapshot. 
4. I guess that if the table definition changes (from the ETL) I'll have to modify the entity and each time it will generate a new migration for the change when I run `mikro-orm migration:create`

**Reply by B4nan** [reply]
> I guess this is because the table doesn't exist in the snapshot.

Yes, sounds like a migration for it was probably ignored (manually removed from the diff) already by you some time ago, otherwise this shouldn't have happened. Unless it's part of a schema that is ignored (either implicitly or explicitly), but since you mentioned you use mysql, schemas shouldn't be relevant, that's only postgres feature.

You can always remove your snapshot and let it regenerate based on the actual schema, maybe you just need to do that?

Also, what table are we talking about? The query that gets existing tables from the database is [defined here](https://github.com/mikro-orm/mikro-orm/blob/master/packages/mysql/src/MySqlSchemaHelper.ts#L36), you can try to check if it returns your table too.
